clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -352,35 +353,36 @@ bool DeclRewriter::areDeclarationsOnSameLine(const DAndReplace &N1, const DAndRe
   return false;
 }
 
-bool DeclRewriter::isSingleDeclaration(const DAndReplace &N) {
-  DeclStmt *Stmt = N.Statement;
+bool DeclRewriter::isSingleDeclaration(VarDeclReplacement *N) {
+  DeclStmt *Stmt = N->getStatement();
   if (Stmt == nullptr) {
-    auto &VDGroup = GP.getVarsOnSameLine(N.getDecl<VarDecl>());
+    auto &VDGroup = GP.getVarsOnSameLine(N->getDecl());
     return VDGroup.size() == 1;
   } else {
     return Stmt->isSingleDecl();
   }
 }
 
-void DeclRewriter::getDeclsOnSameLine(const DAndReplace &D,
+void DeclRewriter::getDeclsOnSameLine(VarDeclReplacement *D,
                                       std::set<Decl *> &Decls) {
-  if (D.Statement != nullptr)
-    Decls.insert(D.Statement->decls().begin(), D.Statement->decls().end());
+  if (D->getStatement() != nullptr)
+    Decls.insert(D->getStatement()->decls().begin(), D->getStatement()->decls().end());
   else
-    Decls.insert(GP.getVarsOnSameLine(D.getDecl<VarDecl>()).begin(),
-                 GP.getVarsOnSameLine(D.getDecl<VarDecl>()).end());
+    Decls.insert(GP.getVarsOnSameLine(D->getDecl()).begin(),
+                 GP.getVarsOnSameLine(D->getDecl()).end());
 }
 
-SourceLocation DeclRewriter::deleteAllDeclarationsOnLine(const DAndReplace &DR) {
-  if (DeclStmt *Stmt = DR.Statement) {
+SourceLocation DeclRewriter::deleteAllDeclarationsOnLine
+    (VarDeclReplacement *DR) {
```

Comment:
The formatting of `(VarDeclReplacement *DR)` on a line on its own is a bit weird.  Is this per the standard style guide? I might have thought you'd put the left paren on the line above this one.